vscode/src/azure/treeView.ts

```diff
@@ -131,6 +131,8 @@ export type Job = {
     | "Finishing"
     | "Cancelled";
   outputDataUri?: string;
+  count: number;
+  shots: number;
```

Comment:
Are both of these always present, or mutually exclusive? If not both always present, one or both should be marked optional.

Reply:
In practice both were always present but I think it's because I only ever test where we've submitted jobs from VS Code where we filled in both fields. I don't think we should trust these to be there for jobs other clients may have submitted.

I changed these to be optional and just overall improved the handling to better sanitize the data coming from the service.